### PR TITLE
feat: customize smtp and white label email templates 

### PR DESCRIPTION
**Pull request by abuaboud**
## What does this PR do?

Fixes # (issue)

### CODE REVIEW

File: packages/backend/src/app/ee/helper/email/email-service.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        const edition = getEdition()
+        if (![ApEdition.CLOUD].includes(edition)) {
+            return
+        }
+        const project = await projectService.getOne(projectId)
+        assertNotNullOrUndefined(project, 'project')
+        if (!isNil(project.platformId)) {
+            // Don't Inform Project Owners, as it breaks white labeling
+            return
+        }
```
Comment (khaledmashaly) [inline packages/backend/src/app/ee/helper/email/email-service.ts:51]: This condition statement is confusing, How does checking whether `project.platformId` exists relates to a user being a platform owner?!

File: packages/backend/src/app/ee/helper/email/email-service.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    })
+    const templateToSubject = {
+        'invitation-email': 'You have been invited to a team',
+        'quota-50': '[ACTION REQUIRED] 50% of your Activepieces tasks are consumed',
+        'quota-90': '[URGENT] 90% of your Activepieces tasks are consumed',
+        'quota-100': '[URGENT] 100% of your Activepieces tasks are consumed',
+    }
+
+    await transporter.sendMail({
+        from: `${platform?.name} <${platform?.smtpSenderEmail ?? 'notifications@activepieces.com'}>`,
```
Comment (khaledmashaly) [inline packages/backend/src/app/ee/helper/email/email-service.ts]: I think let's have a default sender name? like `${platform?.name ?? 'Activepieces' }` maybe?

File: packages/backend/test/integration/ee/platform/platform.test.ts [hunk 3]
```diff
@@ -45,14 +45,22 @@ describe('Platform API', () => {
...
                     filteredPieceNames: ['updated filtered piece names'],
                     filteredPieceBehavior: 'ALLOWED',
+                    smtpHost: 'updated smtp host',
+                    smtpPort: 123,
+                    smtpUser: 'updated smtp user',
+                    smtpPassword: 'updated smtp password',
+                    smtpSenderEmail: 'updated smtp sender email',
+                    smtpUseSSL: true,
+                    privacyPolicyUrl: 'updated privacy policy url',
+                    termsOfServiceUrl: 'updated terms of service url',
```
Comment (khaledmashaly) [inline packages/backend/test/integration/ee/platform/platform.test.ts:55]: I think the new props need to be added to the `createMockPlatform` method

**Review by khaledmashaly — COMMENTED** [review]
Long-awaited clean-up, Excellent! :star_struck: 
Please see my comments!

File: packages/backend/src/app/ee/helper/email/email-service.ts (hunk 1 above)
Comment (abuaboud) [reply packages/backend/src/app/ee/helper/email/email-service.ts:51]: Changed

**Pull request merged**